### ISSUE TITLE
nn.MultiheadAttention is called with the default need_weights=True, which disables the fused attention kernel

### ISSUE THREAD

**Issue by harshal-96** [lyuwenyu/RT-DETR#678]
Every `nn.MultiheadAttention` call site in the PyTorch code throws away the attention
weights:

```python
src, _     = self.self_attn(q, k, value=src,    attn_mask=src_mask)
target2, _ = self.self_attn(q, k, value=target, attn_mask=attn_mask)
```

but none of them pass `need_weights=False`, and it defaults to `True`.

`F.multi_head_attention_forward` can't return a per-head attention matrix out of the
fused kernel, so `need_weights=True` forces the unfused path: separate QK^T, softmax
and AV matmuls, plus materialising the whole `[B, H, N, N]` attention tensor, which
the caller then discards on the same line. With `need_weights=False` the same call
goes through `scaled_dot_product_attention`.

### Repro

```python
import torch, torch.nn as nn

mha = nn.MultiheadAttention(256, 8, dropout=0.0, batch_first=True).cuda().eval()
x   = torch.randn(1, 300, 256, device='cuda')

with torch.inference_mode():
    o_t, w = mha(x, x, x)
    o_f, _ = mha(x, x, x, need_weights=False)

print(w.shape)                         # (1, 300, 300), allocated then thrown away
print((o_t - o_f).abs().max().item())  # 8.9e-08
```

That module issues 9 CUDA kernels with the default and 3 without. In the decoder layer
as a whole (attention + residual + norm) it's 13 vs 6.

### Fix

```diff
-        src, _ = self.self_attn(q, k, value=src, attn_mask=src_mask)
+        src, _ = self.self_attn(q, k, value=src, attn_mask=src_mask, need_weights=False)
```

Five call sites:

- `rtdetr_pytorch/src/zoo/rtdetr/hybrid_encoder.py:148`
- `rtdetr_pytorch/src/zoo/rtdetr/rtdetr_decoder.py:206`
- `rtdetrv2_pytorch/src/zoo/rtdetr/hybrid_encoder.py:148`
- `rtdetrv2_pytorch/src/zoo/rtdetr/rtdetr_decoder.py:206`
- `rtdetrv2_pytorch/src/zoo/rtdetr/rtdetrv2_decoder.py:216`

### Numbers

RTX 4070 Laptop, torch 2.12+cu126, batch 1, 640x640, `rtdetrv2_r50vd`, `deploy()`,
A/B interleaved, this change only: eager 27.0 ms -> 24.9 ms. That's 3-8% depending on
the run; laptop clocks drift, so a single decimal figure would be misleading.

Accuracy is unchanged on all 5000 COCO val2017 images, measured with exactly this
change and nothing else:

| | AP | AP50 | AP75 | APs | APm | APl | AR100 |
|---|---|---|---|---|---|---|---|
| before | 0.5341 | 0.7157 | 0.5747 | 0.361 | 0.579 | 0.708 | 0.7255 |
| after | 0.5341 | 0.7157 | 0.5747 | 0.361 | 0.579 | 0.708 | 0.7256 |

Numerics only move at kernel-reassociation level, 1.5e-3 relative on encoder memory,
and 99.7% of the selected queries are identical.

Training is affected as well as inference, since `need_weights=False` routes through
SDPA and its backward runs during training. Forward with denoising plus the criterion,
then backward, gives a finite loss and the same gradient flow as before (52/52
backbone, 129/129 encoder, 154/194 decoder params with non-zero grads).

One result that looks alarming but isn't: on a `torch.randn` input the final logits
can differ by whole units between the two versions. The Hungarian matcher and the
top-k query selection are both discrete, so near-tied scores get assigned differently
under a 1e-6 change. On real images AP is unchanged, which is the test that counts.

Replacing the modules outright with a hand-written SDPA implementation that packs Q
and K into one GEMM (both call sites pass `q is k`) only bought another ~0.7% over the
one-argument change, so the patch just adds the keyword.
